mitodl/mitxpro
# feat: add vat number

#### Pre-Flight checklist


#### What are the relevant tickets?
(Required)
Closes https://github.com/mitodl/mitxpro/issues/2762

#### What's this PR do?
(Required)
Adds a new field `vat_id` in the legal address model. Users can add VAT ID while creating an account and edit from the profile page. Also, Adds it to the receipts.

#### How should this be manually tested?
(Required)

- Create a new account and verify that you can see a field toggle for VAT ID under Full Name.
- Click `Add VAT ID` and verify that you can see the VAT ID Field.
- Add some VAT ID and complete account creation.
- Visit the profile page and see that you can view the VAT ID in profile page.
- Click to edit the profile page. VAT ID Field should be visible in case VAT ID is added otherwise you will see a `Add VAT ID` toggle.
- Enroll in a course and complete payment/ use Coupon.
- Verify that VAT ID is added in Email Receipt and Dashboard Receipt.
- Verify that VAT ID is synced with HubSpot


#### Screenshots
![Screen Shot 2023-12-04 at 4 20 06 PM](https://github.com/mitodl/mitxpro/assets/52656433/74750e50-03fd-4770-a424-67a3a490fd6c)
![Screen Shot 2023-12-04 at 4 20 11 PM](https://github.com/mitodl/mitxpro/assets/52656433/4a691913-11eb-4f18-b4ca-4f937e6952c9)
![Screen Shot 2023-12-04 at 4 20 54 PM](https://github.com/mitodl/mitxpro/assets/52656433/b820cf02-e7fc-4428-a885-41c605cbfff6)
![Screen Shot 2023-12-04 at 4 21 22 PM](https://github.com/mitodl/mitxpro/assets/52656433/5623ab70-a790-4c3f-9260-01d9bfcd1d44)
![Screen Shot 2023-12-04 at 4 21 49 PM](https://github.com/mitodl/mitxpro/assets/52656433/25edac3e-0b99-4858-8e72-8d70c76b3d7a)
![Screen Shot 2023-12-04 at 4 22 25 PM](https://github.com/mitodl/mitxpro/assets/52656433/db5ceb7f-2ffb-4100-bda0-52452cda8ca8)
![Screen Shot 2023-12-04 at 4 22 55 PM](https://github.com/mitodl/mitxpro/assets/52656433/61b3a6a4-c687-4611-817f-e392fbdf15a1)
